core/src/main/scala/org/apache/spark/deploy/rest/RestSubmissionClient.scala

```diff
@@ -413,6 +412,15 @@ private[spark] object RestSubmissionClient {
     val mainClass = args(1)
     val appArgs = args.slice(2, args.size)
     val conf = new SparkConf
-    run(appResource, mainClass, appArgs, conf)
+    val env = filterSystemEnvironment(sys.env)
+    run(appResource, mainClass, appArgs, conf, env)
+  }
+
+  /**
+   * Filter non-spark environment variables from any environment.
+   */
+  def filterSystemEnvironment(env: Map[String, String]): Map[String, String] = {
```

Comment:
There is a test for it, so I'd need to make it visible to the test somehow or remove the test.

I've changed it to protected[rest]. LMK if you have an alternate solution or would like me to make it private and remove the test.